Sources/BraintreeCard/CreditCardGraphQLBody.swift

```diff
@@ -0,0 +1,223 @@
+// swiftlint:disable all
+import Foundation
+
+struct CreditCardGraphQLBody: Encodable {
+
+    // MARK: - Internal Properties
+    
+    var variables: Variables
+    var query: String
+    var operationName: String
+    
+    init(card: BTCard) {
+        self.variables = Variables(card: card)
+        self.query = Self.cardTokenizationGraphQLMutation(authenticationInsightRequested: card.authenticationInsightRequested)
+        self.operationName = "TokenizeCreditCard"
+    }
+
+    struct Variables: Encodable {
+
+        var input: Input
+        
+        init(card: BTCard) {
+            self.input = Input(card: card)
+        }
+        
+        struct Input: Encodable {
+
+            var creditCard: CreditCard
+            var options: Options
+            var authenticationInsightInput: AuthenticationInsightInput?
+
+            init(card: BTCard) {
+                self.creditCard = CreditCard(card: card)
+                self.options = Options(validate: card.shouldValidate)
+                
+                if card.authenticationInsightRequested {
+                    self.authenticationInsightInput = AuthenticationInsightInput(card: card)
+                }                
+            }
+            
+            struct CreditCard: Encodable {
+                var billingAddress: BillingAddress?
+                var number: String?
+                var expirationMonth: String?
+                var cvv: String?
+                var options: Options?
+                var expirationYear: String?
+                var cardholderName: String?
+
+                init(card: BTCard) {                    
+                    if card.firstName != nil {
+                        self.billingAddress = BillingAddress(card: card)
+                    }
```

Comment:
Why do we evaluate whether the firstName is not null to create the BillingAddress? I believe that BillingAddress is optional and does not depend on the firstName. What do you think?
```suggestion
                    self.billingAddress = BillingAddress(card: card)
```

Reply:
This was the original logic. If we don't need it that would be fantastic. @jaxdesmarais can you weigh in on whether this is a business rule?

Reply:
I don't see where the logic to check the first name existed for billing address unless I am missing it: https://github.com/braintree/braintree_ios/blob/main/Sources/BraintreeCard/BTCard.swift. Looks like that was changed in this PR.

Reply:
I think you're right. I'm not sure why I remembered some logic that encased the BillingInfo...

Reply:
Okay, I think we can remove this condition here in `func encode`
```
                 if let billingAddress {
                        try container.encodeIfPresent(billingAddress, forKey: .billingAddress)
                    }
```
But we still need to keep a conditional check here 
```
init(card: BTCard) {
    if [someCondition]                 
     self.billingAddress = BillingAddress(card: card)
```

Doing so will keep this original logic intact for not sending an empty BillingAddress json object 
```
          if !billingAddressDictionary.isEmpty {
            cardDictionary["billingAddress"] = billingAddressDictionary
        }
```

Reply:
Gotcha, that makes sense to me. Thanks for tracking that down. Maybe here we can do `if !card.billingAddress. isEmpty` to match the rest of the object? What do you think?

Reply:
Ah you may be right, I don't think it exists yet. Maybe then doing this makes the most sense:
```swift
if let billingAddress {
    try container.encodeIfPresent(billingAddress, forKey: .billingAddress)
}
```
I think there are cases when merchants pass a billing address but don't pass a first name, and with the logic as is we just wouldn't pass the other details along even if they existed if no firstName exists.